File: core/src/main/java/com/uiptv/util/EmbeddedPlayerWideViewUtil.java
package com.uiptv.util;

import com.uiptv.model.Configuration;
import com.uiptv.service.ConfigurationService;

public final class EmbeddedPlayerWideViewUtil {
    private EmbeddedPlayerWideViewUtil() {
    }

    public static boolean isWideViewEnabled() {
        Configuration configuration = ConfigurationService.getInstance().read();
        return configuration != null && configuration.isEmbeddedPlayer() && configuration.isWideView();
    }
}

File: server/src/main/kotlin/com/uiptv/util/EmbeddedPlayerWideViewUtil.kt
package com.uiptv.util

import com.uiptv.service.ConfigurationService

object EmbeddedPlayerWideViewUtil {
    private var configurationServiceProvider: () -> ConfigurationService = { ConfigurationService }

    @JvmStatic
    fun configureDependencies(configurationService: ConfigurationService) {
        this.configurationServiceProvider = { configurationService }
    }

    @JvmStatic
    fun resetDependencies() {
        configurationServiceProvider = { ConfigurationService }
    }

    @JvmStatic
    fun isWideViewEnabled(): Boolean {
        val configuration: com.uiptv.model.Configuration? = configurationServiceProvider().read()
        if (configuration == null) {
            return false
        }
        return configuration.embeddedPlayer && configuration.wideView
    }
}
